### PR TITLE
fix: suppress compiler warnings

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

Fixed warnings for Rolling.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tier4_vehicle_rviz_plugin/src/tools/steering_angle.cpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ void SteeringAngleDisplay::update(float wall_dt, float ros_dt)
   // ((property_handle_angle_scale_->getFloat() * (msg_ptr->data / M_PI) * -180.0));
   int handle_image_width = handle_image_.width(), handle_image_height = handle_image_.height();
   QPixmap rotate_handle_image;
-  rotate_handle_image = handle_image_.transformed(rotation_matrix);
+  rotate_handle_image = handle_image_.transformed(QTransform(rotation_matrix));
```
Comment (kenji-miyake) [inline common/tier4_vehicle_rviz_plugin/src/tools/steering_angle.cpp:133]: `transformed(QMatrix)` is deprecated.

File: localization/pose_initializer/src/pose_initializer_core.cpp [hunk 2]
```diff
@@ -188,7 +192,7 @@ bool PoseInitializer::getHeight(
     input_pose_msg.pose.pose.position.z);
 
   if (map_ptr_) {
-    tf2::Transform transform;
+    tf2::Transform transform{tf2::Quaternion{}, tf2::Vector3{}};
```
Comment (kenji-miyake) [inline localization/pose_initializer/src/pose_initializer_core.cpp:195]:
```
error: 'transform.tf2::Transform::m_origin.tf2::Vector3::m_floats[1]' may be used uninitialized in this function [-Werror=maybe-uninitialized]
```

File: sensing/probabilistic_occupancy_grid_map/src/pointcloud_based_occupancy_grid_map/occupancy_grid_map.cpp [hunk 3]
```diff
@@ -301,8 +306,6 @@ void OccupancyGridMap::updateWithPointCloud(
   // Third step: Overwrite occupied cell
   for (size_t bin_index = 0; bin_index < obstacle_pointcloud_angle_bins.size(); ++bin_index) {
     auto & obstacle_pointcloud_angle_bin = obstacle_pointcloud_angle_bins.at(bin_index);
-    auto & raw_pointcloud_angle_bin = raw_pointcloud_angle_bins.at(bin_index);
```
Comment (kenji-miyake) [inline sensing/probabilistic_occupancy_grid_map/src/pointcloud_based_occupancy_grid_map/occupancy_grid_map.cpp:304]:
```
error: unused variable 'raw_pointcloud_angle_bin' [-Werror=unused-variable]
  309 |     auto & raw_pointcloud_angle_bin = raw_pointcloud_angle_bins.at(bin_index);
```

File: simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp [hunk 4]
```diff
@@ -273,5 +273,5 @@ const std::string VEHICLE_MODEL_LIST[] = {   // NOLINT
 };
 // clang-format on
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
```
Comment (kenji-miyake) [inline simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp:276]:
```
error: 'constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()' is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Werror=deprecated-declarations]
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
```

**Review by wep21 — APPROVED** [review]
LGTM

**Pull request merged**